### PR TITLE
experiment with adding Profiling to measure Gatsby builds

### PR DESCRIPTION
**Pull request by AbhiPrasad**
@JonasBa Some stuff I tried out the other week.

Putting profiling into bundlers/build tooling is very interesting, and could be a nice immediate value add for us esp since we are thinking about doing more work with bundlers.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,6 +8,8 @@
     "@mdx-js/react": "^v1.6.22",
     "@sentry-internal/global-search": "0.0.43",
     "@sentry/gatsby": "^7.15.0",
+    "@sentry/node": "^7.15.0",
+    "@sentry/profiling-node": "^0.0.0-alpha.1",
```
Comment (JonasBa) [inline package.json:12]: Arf, can you make sure you install 0.0.0-alpha.6 or bump this to .7 (I'll cut a release this morning with some changes)

Comment (AbhiPrasad) [reply package.json:12]: yes can do!

**Comment by JonasBa** [comment]
@AbhiPrasad that would be great as we can add extra tags based off the hook we are profiling in and have better visibility into what step the bundler is executing.

**Comment by AbhiPrasad** [comment]
https://sentry.io/organizations/sentry/profiling/?project=5266138&statsPeriod=24h

Thinking of merging this in, so we collect some profiles over time. Not sure why the build stuff is not working 🤔

File: gatsby-node.js [hunk 2]
```diff
@@ -1,4 +1,90 @@
...
 exports.createSchemaCustomization = require("./src/gatsby/createSchemaCustomization").default;
+
+Sentry.configureScope(scope => {
+  const id = process.env.GITHUB_RUN_ID;
+  if (id) {
+    scope.setUser({ id });
+  }
+});
+
+function startFromPrevTransaction(newTransactionContext) {
```
Comment (AbhiPrasad) [inline gatsby-node.js:31]: the api making me have to add this helper

scream-toad

File: package.json [hunk 3]
```diff
@@ -8,6 +8,8 @@
     "@mdx-js/react": "^v1.6.22",
     "@sentry-internal/global-search": "0.0.43",
     "@sentry/gatsby": "^7.15.0",
+    "@sentry/node": "^7.15.0",
+    "@sentry/profiling-node": "^0.0.1",
```
[the remark this answers is not in the document]
Comment (AbhiPrasad) [reply package.json]: done!

**Pull request merged**